Fix dashboard tests

notoraptor opened the pull request:
# Description

Try to fix dashboard tests.

# Changes

- Update playwright
- Handle new pandas version 2.0.0

# Checklist

## Tests
- [ ] I added corresponding tests for bug fixes and new features. If possible, the tests fail without the changes
- [ ] All new and existing tests are passing (`$ tox -e py38`; replace `38` by your Python version if necessary)

## Documentation
- [ ] I have updated the relevant documentation related to my changes

## Quality
- [ ] I have read the [CONTRIBUTING](https://github.com/Epistimio/orion/blob/develop/CONTRIBUTING.md) doc
- [ ] My commits messages follow [this format](https://chris.beams.io/posts/git-commit/)
- [ ] My code follows the style guidelines (`$ tox -e lint`)

notoraptor committed:
Update playwright dependencies.

notoraptor committed:
- run yarn ci-check
- check ulimit on playwright CI

notoraptor committed:
Try to set ulimit to greater values

notoraptor committed:
Try sudo to set ulimit

notoraptor committed:
Try to increase user watchers

notoraptor committed:
Try to run tests in sequence, not in parallel, to reduce number of connections to mongodb

notoraptor committed:
Run tests in parallel again
Try again to set a ulimit
Try to get orion server log file

notoraptor committed:
Try to set ulimit without sudo

notoraptor committed:
Remove ulimit
Try to update mongodb to version 6.0

notoraptor committed:
Try to configure mongodb as done in github workflow `build.yml`

notoraptor committed:
Try to get artifact

notoraptor committed:
Force pandas version to 1.4.4

notoraptor committed:
Update to support pandas 2.0.0

notoraptor committed:
Tests passed. Re-simplify workflow (seems unnecessary to install mongoshell)

notoraptor committed:
Do not specify pandas version, so that appropriate version is selected for each python version
pandas 2.0 is available only for python >= 3.8
pandas up to 1.3.5 is available for python <= 3.7

notoraptor commented:
Hi @bouthilx ! It seems I have fixed the issue related to failing dashboard tests on CI!

Ultimately, the solution was to handle both new and old versions of pandas by specify parameter `group_keys=False` when using `groupby()`: https://github.com/Epistimio/orion/pull/1096/files#diff-ad4314cde86cb004bc4eeacc08ea818db0887a0e74888f4064ad384ddd9a48d7

Before:
```python
return trials.groupby(group_by).apply(rank)
```

After:
```python
return trials.groupby(group_by, group_keys=False).apply(rank)
```

`group_keys` argument is already available in older versions of pandas, but it seems pandas 2.0.0 changes its behavior. So, to keep old behavior, a warning from pandas 1.3.5 (version just before 2.0.0) suggests to use `group_keys=False`:

```
/home/stevenbocco/mila/dev/git/epistimio/orion/src/orion/analysis/base.py:107: FutureWarning:

Not prepending group keys to the result index of transform-like apply. In the future, the group keys will be included in the index, regardless of whether the applied function returns a like-indexed object.
To preserve the previous behavior, use

	>>> .groupby(..., group_keys=False)

To adopt the future behavior and silence this warning, use 

	>>> .groupby(..., group_keys=True)
```

This seems sufficient to fix tests, no need to change ulimit.

So, dashboard tests pass again! But now I have issues with build tests and long algos.

- For build tests, I don't yet know how to fix them: https://github.com/Epistimio/orion/actions/runs/4737274539/jobs/8409954043?pr=1096
- For long algos, errors occur on Python 3.7: https://github.com/Epistimio/orion/actions/runs/4737274539/jobs/8409954604?pr=1096

About long algo issues, it seems related to package `ax-platform` which uses `typeguard` with a new version that causes backward incompatibilities. I found this ticket: https://github.com/facebook/Ax/pull/1510

On Python 3.8, new ax package seems to have fixed the issue by forcing a typeguard version in dependencies: https://github.com/facebook/Ax/blob/0.3.1/setup.py#L31

But, on Python 3.7, latest supported ax package is 0.2.6, from August 2022, and I guess it is not updated and relies on `typeguard` by default, without hardcoded version: https://github.com/facebook/Ax/blob/0.2.6/setup.py#L29

So, I don't know what strategy to adopt. Should I try to "force" orion to install `typeguard=2.13.3` for Python <= 3.7 ?

bouthilx commented:
Python 3.7 end of life is scheduled for 2023-06-27. I think we can safely drop support for 3.7 in next release, so now in `develop`.

bouthilx commented:
The failed test in the build tests look like a glitch. I relaunched them.

notoraptor committed:
Remove unnecessary comment about how code was adjusted to support current version of Pandas 2.0.0.
Removed comment:
  Added group_keys=False to keep previous behavior from pandas < 2 into pandas >= 2.
  Fix based on warning message from pandas 1.5.3:
    Not prepending group keys to the result index of transform-like apply.
    In the future, the group keys will be included in the index,
    regardless of whether the applied function returns a like-indexed object.
    To preserve the previous behavior, use
    >>> .groupby(..., group_keys=False)
    To adopt the future behavior and silence this warning, use
    >>> .groupby(..., group_keys=True)
